hazelcast/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -519,21 +512,21 @@ private boolean isResourceMapExpired(long creationTime) {
 
     Set<Long> getAllJobIds() {
         Set<Long> ids = new HashSet<>();
-        ids.addAll(jobRecords.keySet());
-        ids.addAll(jobResults.keySet());
+        ids.addAll(jobRecords().keySet());
+        ids.addAll(jobResults().keySet());
         return ids;
     }
 
     public Collection<JobRecord> getJobRecords() {
```

Comment:
This is called from `com.hazelcast.jet.impl.JobCoordinationService#doScanJobs` which is scheduled to be called every 5 ms by default, so the map will be created for tests taking longer than few ms. We should check if the map exists at all and return empty collection if not.

Reply:
currently, all members start scanning for jobs after a node starts, but only the master member is doing the actual scanning. Other members are not doing anything but keep scheduling the scan so that when it becomes master starts the jobs. 

I've removed the starting of the scan after the node starts and put it to `submitJob`. But this means the scan will be only on the master member. I've also added a membership listener to listen the member removed evets and check if the current member is the master and start scanning. 

I've also guarded startScanningForJobs with an AtomicBoolean to not start multiple scans.